dataset_loaders/videos/movingMNIST.py

```diff
@@ -0,0 +1,222 @@
+        length = self.seq_length + 1
+        canvas_size = self.image_size_ - np.max(self.digits_sizes_)
+
+        # Initial position uniform random inside the box.
+        y = self._rng.rand(batch_size)
+        x = self._rng.rand(batch_size)
+        # Choose speed.
+        speed = self.digits_speed
+        if isinstance(self.digits_speed, float):
+            speed = [self.digits_speed for el in range(batch_size)]
+        elif self.digits_speed == 'random':
+            speed = self._rng.rand(batch_size)
+        v_y = np.asarray(speed)  # np.sin(theta)
+        v_x = np.asarray(speed)  # np.cos(theta)
+
+        start_y = np.zeros((length, batch_size))
+        start_x = np.zeros((length, batch_size))
+        # Random directions
+        y_rand = np.asarray([self._rng.uniform(-1, 1) for _ in
+                             range(batch_size)])
+        x_rand = np.asarray([self._rng.uniform(-1, 1) for _ in
+                             range(batch_size)])
+
+        # For each digit for each frame compute the decision to change
+        # direction following a binomial distribution
+        change_dir = []
+        for i in range(self.num_digits_):
+            sampling = [self._rng.binomial(
+                1, self.change_dir_prob[i]) for _ in range(length)]
+            change_dir.append(sampling)
+        change_dir = zip(*change_dir)
+
+        for i in xrange(length):
+
+            # Which digits change the direction in this frame
+            c_digits = np.where(change_dir[i])
+            # Number of digits changing direction in this frame
+            n_digits = len(c_digits)
+            np.put(y_rand, c_digits,
+                   [self._rng.uniform(-1, 1) for _ in range(n_digits)])
+            np.put(x_rand, c_digits,
+                   [self._rng.uniform(-1, 1) for _ in range(n_digits)])
+            # Take a step along velocity.
+            y += v_y * y_rand  # + self.step_length_
+            x += v_x * x_rand  # + self.step_length_
+            # Bounce off edges.
+            for j in xrange(batch_size):
+                if x[j] <= 0:
+                    x[j] = 0
+                    v_x[j] = -v_x[j]
+                if x[j] >= 1.0:
+                    x[j] = 1.0
+                    v_x[j] = -v_x[j]
+                if y[j] <= 0:
+                    y[j] = 0
+                    v_y[j] = -v_y[j]
+                if y[j] >= 1.0:
+                    y[j] = 1.0
+                    v_y[j] = -v_y[j]
+            start_y[i, :] = y
+            start_x[i, :] = x
+
+        # Scale to the size of the canvas.
+        start_y = (canvas_size * start_y).astype(np.int32)
+        start_x = (canvas_size * start_x).astype(np.int32)
+        return start_y, start_x
+
+    def get_batch(self, verbose=False):
+        start_y, start_x = self.get_random_trajectory(
+            self.batch_size_ * self.num_digits_)
+
+        # minibatch data
+        # if self.background_ == 'zeros':
```

Comment:
Do not leave commented out code!
Either use it or remove it. The background doesn't seem a bad thing to have, if that works can you uncomment the code?